dashboards-observability/public/components/visualizations/charts/data_table/data_table.tsx

```diff
@@ -3,102 +3,487 @@
+          animateRows={true}
+          rowSelection="multiple"
+          enableRangeSelection={true}
+          pagination={true}
+          paginationPageSize={pageSize}
+          paginationNumberFormatter={paginationNumberFormatter}
+          suppressPaginationPanel={true}
+          rowHeight={selectedRowDensity.height}
+        />
+      </div>
+      <GridFooter
+        onPageSizeChanged={onPageSizeChanged}
+        pageSize={pageSize}
+        activePage={activePage}
+        goToPage={goToPage}
+        pageCount={pageCount}
+      />
+      {isFullScreen && (
+        <CustomOverlay>
+          <EuiFlexGroup direction="column">
+            <EuiFlexItem>
+              <div style={{ height: '500px' }}>
+                <AgGridReact
+                  ref={gridRefFullScreen}
+                  rowData={raw_data}
+                  columnDefs={columns}
+                  defaultColDef={defaultColDef}
+                  animateRows={true}
+                  rowSelection="multiple"
+                  enableRangeSelection={true}
+                  pagination={true}
+                  // domLayout={'autoHeight'}
+                  paginationPageSize={pageSize}
+                  paginationNumberFormatter={paginationNumberFormatter}
+                  suppressPaginationPanel={true}
+                  rowHeight={selectedRowDensity.height}
+                />
+              </div>
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <GridFooter
+                onPageSizeChanged={onPageSizeChanged}
+                pageSize={pageSize}
+                activePage={activePage}
+                goToPage={goToPage}
+                pageCount={pageCount}
+              />
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </CustomOverlay>
+      )}
+    </>
+  );
+};
 
-  const renderCellValue = useMemo(() => {
-    return ({ rowIndex, columnId }) => {
-      let adjustedRowIndex = rowIndex;
+const CustomOverlay = ({ children }: { children: any }) => {
+  return (
+    <div
+      className={
+        uiSettingsService.get('theme:darkMode')
+          ? 'custom-overlay custom-overlay-dark'
+          : 'custom-overlay custom-overlay-light'
+      }
+    >
+      {children}
+    </div>
+  );
+};
 
-      // If we are doing the pagination (instead of leaving that to the grid)
-      // then the row index must be adjusted as `data` has already been pruned to the page size
-      adjustedRowIndex = rowIndex - pagination.pageIndex * pagination.pageSize;
+const GridHeader = ({
+  isFullScreen,
+  setIsFullScreenHandler,
+  selectedRowDensity,
+  selectDensityHandler,
+  columnVisiblityHandler,
+  columns,
+  columnVisibility,
+}: {
+  isFullScreen: boolean;
+  setIsFullScreenHandler: (v: boolean) => void;
+  selectedRowDensity: any;
+  selectDensityHandler: (v: any) => void;
+  columnVisiblityHandler: (visible: boolean, feild: any) => void;
+  columns: any;
+  columnVisibility: any;
+}) => {
+  return (
```

Comment:
You can define something like `itemStyle = { maxWidth: '150px' }` here and use it `<EuiFlexItem style={itemStyle}>`. This way, in case you have to update the hardcoded width to say 200px, then it is going to be less changes at that time.